File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
import java.util.*;

class Solution {
    
    static TreeSet<String> make = new TreeSet<>();
    static String vowel[] = {"A", "E", "I", "O", "U"};
    static boolean visited[] = new boolean[5];
    
    public int solution(String word) {
        
        for(int i=1; i<=5; i++){
            permutation("", i);
        }
        
        return make.headSet(word).size() +1;
    }
    
    public void permutation(String word, int len){
        
        if(word.length() == len){
            make.add(word);
            return;
        }
        
        for(int i=0; i<vowel.length; i++){
            word = word + vowel[i];
            permutation(word, len);
            word = word.substring(0, word.length()-1);
        }
        
    }

}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    
    const vowel = ["A", "E", "I", "O", "U"]
    const count = [781, 156, 31, 6, 1]
     
    var answer = word.length;
    
    for(let i=0; i<word.length; i++){
        let index = vowel.indexOf(word[i]);
        answer += count[i]*index;
    }
  
    return answer;
}